os/arch/arm/src/common/up_allocateheap.c

```diff
@@ -200,3 +204,29 @@ void up_allocate_kheap(FAR void **heap_start, size_t *heap_size)
 	*heap_size = (uint32_t)((uintptr_t)__usram_segment_start__) - (uint32_t)(*heap_start);
 }
 #endif
+
+/****************************************************************************
+ * Name: up_addregion
+ ****************************************************************************/
+#if CONFIG_MM_REGIONS > 1
+void up_addregion(void)
+{
+	char *mem_start = CONFIG_HEAPx_BASE;
+	char *mem_size = CONFIG_HEAPx_SIZE;
+
+	while(*mem_start && *mem_size) {
+		kumm_addregion((void *)strtol(mem_start, NULL, 16), (size_t)strtol(mem_size, NULL, 0));
```

Comment:
1. for mem_size, is 0 right for third arg?
2. Don't you need to split by ','?
3. Don't you need to set endptr??